### PR TITLE
Multi-select

### PR DESCRIPTION
**Pull request by matthrice**
Supports selecting multiple features in the collection. An array of indexes is given as props instead of a single number. Selected features may be modified while sharing selection, but drawing lines/polygons can only occur when 1 or 0 features are selected. 

#### EditableFeatureCollection:
* add `featureIndex` property to `editHandleInfo` to distinguish which feature is being edited while multiple features are selected
* **Note:** this causes the edit handles test suite to fail, if this were to be merged in as is I'd have to update those tests

#### EditableGeoJsonLayer
* update all instances of `selectedFeature` and `selectedFeatureIndex` to arrays `selectedFeatures` and `selectedFeatureIndexes`

#### Examples
* Updated the deck example to use `selectedFeatureIndexes` 
* Provided a new example to show the functionality of multi-select

### CODE REVIEW

File: examples/data/austin-geojson.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
+{
```
Comment (georgios-uber) [inline examples/data/austin-geojson.json]: Can we use https://github.com/uber-common/deck.gl-data instead?

**Comment by georgios-uber** [comment]
Please **UPDATE DOCS**

File: examples/data/austin-geojson.json (hunk 1 above)
Comment (matthrice) [reply examples/data/austin-geojson.json]: The vancouver data? It's a little large to just demonstrate multi select I think, but I could write another example using it

Comment (georgios-uber) [reply examples/data/austin-geojson.json]: Maybe you can push this data there and use it using Ajax.

**Comment by georgios-uber** [comment]
Seems good.

File: docs/api-reference/layers/editable-geojson-layer.md [hunk 2]
```diff
@@ -103,7 +107,7 @@ The `onEdit` event is the core event provided by this layer and must be handled
 
 * `updatedMode` (String): A suggested value to use for `mode` after the edit is applied. Often this is the same value. But occasionally, an edit will need to transition the layer from one mode to another.
 
-* `updatedSelectedFeatureIndex` (Number): A suggested value to use for `selectedFeatureIndex` after the edit is applied. Often this is the same value. But occasionally, an edit will change the selected feature (e.g. when creating a new feature, it goes from `null` to the index of the newly created feature).
+* `updatedSelectedFeatureIndexes` (Number): A suggested array to use for `selectedFeatureIndexes` after the edit is applied. Often this is the same varray. But occasionally, an edit will change the selected features (e.g. when creating a new feature, it goes from `[]` to an array holding the index of the newly created feature).
```
Comment (supersonicclay) [inline docs/api-reference/layers/editable-geojson-layer.md]: typo: "varry"

File: docs/api-reference/layers/editable-geojson-layer.md [hunk 3]
```diff
@@ -75,20 +75,24 @@ The `mode` property dictates what type of edits the user can perform and how to
...
-  * If no feature is selected, clicking will create a new feature `Point` feature and select it (by passing its index as `updatedSelectedFeatureIndex`).
+  * If no feature is selected, clicking will create a new feature `Point` feature and select it (by passing its index as `updatedSelectedFeatureIndexes`).
 
   * If a `Point` feature is selected, clicking will convert it to a `LineString` and add the clicked position to it.
 
   * If a `LineString` feature is selected, clicking will add the clicked position to it. If that clicked position is the first position, the feature will be converted to a `Polygon` feature.
 
+  * If multiple features are selected, the user will be prevented from drawing
+
 #### `onEdit` (Function, optional)
```
Comment (supersonicclay) [inline docs/api-reference/layers/editable-geojson-layer.md]: Oops, I never documented the `selectedFeatureIndex` prop. This is a good time to fix that and document the `selectedFeatureIndexes` prop.

File: src/lib/layers/editable-geojson-layer.js [hunk 4]
```diff
@@ -147,11 +156,13 @@ export default class EditableGeoJsonLayer extends EditableLayer {
   }
 
   isFeatureSelected(feature: Object) {
-    if (!this.props.data) {
+    // TODO: Upgrade deck to include https://github.com/uber/deck.gl/commit/40ff66ed
```
Comment (supersonicclay) [inline src/lib/layers/editable-geojson-layer.js]: this TODO no longer applies and can be removed

File: src/lib/layers/editable-geojson-layer.js [hunk 5]
```diff
@@ -147,11 +156,13 @@ export default class EditableGeoJsonLayer extends EditableLayer {
...
+    // This will be buggy until then
+    if (!this.props.data || !this.props.selectedFeatureIndexes.length) {
```
Comment (supersonicclay) [inline src/lib/layers/editable-geojson-layer.js]: check `!this.props.selectedFeatureIndexes` before `!this.props.selectedFeatureIndexes.length`

**Pull request merged**